src/presentation/react/core/components/sync-wizard/common/SummaryStep.tsx

```diff
@@ -183,8 +184,19 @@ export const SummaryStepContent = (props: SummaryStepContentProps) => {
             result.match({
                 error: () => snackbar.error(i18n.t("Invalid origin instance")),
                 success: instance => {
+                    Promise.all(
+                        syncRule.metadataSyncAll.map(
+                            async type =>
+                                await compositionRoot.metadata
+                                    .listAll({ type: type as keyof MetadataEntities, fields: { id: true } }, instance)
+                                    .then(metadata => ({
+                                        [type]: metadata.filter(({ id }) => !syncRule.excludedIds.includes(id)).length,
+                                    }))
+                        )
+                    ).then(typeLength => setSyncAllTypesLength(Object.assign({}, ...typeLength)));
+
```

Comment:
As a general rule, we should have a useCase which does the task so you don't do this Promise.all+map+... in the view.